cli/integrity/datapublisher.go

```diff
@@ -18,10 +18,10 @@ type DataPublisherFactory interface {
 	// NewFile creates a new DataPublisher to publish data into a file.
 	NewFile(path string) (DataPublisher, error)
 	// NewEtcd creates a new DataPublisher to publish data into etcd.
-	NewEtcd(etcdcli *clientv3.Client,
-		prefix, key string, timeout time.Duration) (DataPublisher, error)
+	NewEtcd(etcdcli *clientv3.Client, prefix, key string,
+		modRevision int64, timeout time.Duration) (DataPublisher, error)
```

Comment:
I think it's a much better to change the publisher interface to:
```Go
Publish(revision int64, data []byte)
```

In that case we could create a publisher/collector once. And than to do:
```Go
for {
    data := collector.Collect()
    // something
    publisher.Publish(data.Revision, newData)
}
```

Reply:
I've been thinking about this approach, but it seems confusing to me because we force `FileDataPublisher` to satisfy an interface, which it can not implement. 

About structure creations: anyway, these ones are lightweight.

Reply:
It's looks fine for me. FileDataPublisher (or any other publisher) does not supports target revision and we could have an error from it in the case.

Reply:
This approach produces checks in several structures (EtcdAllDataPublisher, TarantoolAllDataPublisher, FileDataPublisher), which clutters up the code, on my taste. But if we are not planning to add some other arguments to publishers, it may be ok.

Reply:
In fact *AllPublishers may check some global revision. So it's ok to return an error if it is unimplemented now.

Reply:
Thank you! Changes are applied.